pkg/kubelet/images/image_manager.go

```diff
@@ -85,7 +87,15 @@ func (m *imageManager) EnsureImageExists(pod *api.Pod, container *api.Container,
 		glog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
 	}
 
-	spec := kubecontainer.ImageSpec{Image: container.Image}
+	// If the image contains no tag or digest, a default tag should be applied.
+	image, err := applyDefaultImageTag(container.Image)
```

Comment:
It seems that when pulling image, they use `ImageSpec.Image` directly https://github.com/kubernetes/kubernetes/blob/b1ee54346347f3675f285d24e302d36980983804/pkg/kubelet/rkt/image.go#L47. In docker, if we use it directly without any tag, it will pull all images with the same name. What will happen for rkt in the same case?

> Instead of letting each runtime sets a different default, it makes more sense for kubelet to set the default to ensure consistent user experience.

Yeah, just don't know whether different runtimes have the same default. For now, it's fine because everyone is using docker image, and the default should all be `:latest` (?).